src/compiler/parser.ts

```diff
@@ -3221,6 +3221,37 @@ module ts {
                     }
                 }
 
+                // If encounter "([", this could be the start of an array binding pattern.
+                // Examples:
+                //      ([ x ]) => { }
+                //      ([ x ])
+                if (second === SyntaxKind.OpenBracketToken) {

```

Comment:
I went ahead and just made OpenBrace/OpenBracket return Tristate.Unknown. The tryParse will already be able to detect this after reading two tokens, so its not much of an added cost to be worth it.